exporters/postgres/cht_sync_collector.yml

```diff
@@ -0,0 +1,75 @@
+collector_name: cht-sync
+
+# Update the schema and table names as needed
+queries:
+  - query_name: couch2pg-query
+    query: |
+      SELECT
+        split_part(seq,'-',1) as sequence,
+        pending as pending,
+        CASE 
+          WHEN updated_at < NOW() - INTERVAL '1 minute' THEN 0
+          ELSE 1
+        END AS liveness,
+        split_part(source,'/',2) as db,
+        split_part(source,'/',1) as cht_instance
+      FROM
+        v1.couchdb_progress
+      WHERE
+        source like '%/%' and
+        seq like '%-%'
+      ORDER BY
+        cht_instance, db
+  - query_name: dbt-latency
+    query: |
+      SELECT
+        EXTRACT(EPOCH FROM(couchdb.latest - dbt_root.latest)) AS dbt_latency
+      FROM
+        (SELECT MAX(saved_timestamp) as latest FROM v1.document_metadata) dbt_root,
+        (SELECT MAX(saved_timestamp) as latest FROM v1.couchdb) couchdb
```

Comment:
I'm getting an error:

```
ts=2024-10-16T22:22:52.357Z caller=klog.go:134 level=error func=Errorf msg="Error gathering metrics: [from Gatherer #1] [job=db_targets,target=db1,collector=cht-sync,query=dbt-latency] pq: relation \"v1.couchdb\" does not exist"
ts=2024-10-16T22:22:53.357Z caller=klog.go:134 level=error func=Errorf msg="Error gathering metrics: [from Gatherer #1] [job=db_targets,target=db1,collector=cht-sync,query=dbt-latency] pq: relation \"v1.couchdb\" does not exist"
ts=2024-10-16T22:22:54.356Z caller=klog.go:134 level=error func=Errorf msg="Error gathering metrics: [from Gatherer #1] [job=db_targets,target=db1,collector=cht-sync,query=dbt-latency] pq: relation \"v1.couchdb\" does not exist"
```

Doing some digging in the actual data in my DB, this might work?

```suggestion
        (SELECT max(v1.couchdb_progress.updated_at) as latest FROM v1.couchdb_progress) couchdb
```

Checking prometheus targets at `http://localhost:9090/targets?search=` I see `http://sql_exporter:9399/metrics` marked as `UP`, and checking that URL I see metrics I'd expect:

```
# HELP couch2pg_progress_pending approximate number of changes left to sync from couch to postgres
# TYPE couch2pg_progress_pending gauge
couch2pg_progress_pending{cht_instance="192-168-68-26.local-ip.medicmobile.org",db="medic",job="db_targets",target="db1"} 0
# HELP couch2pg_progress_sequence current sequence number for couch2pg
# TYPE couch2pg_progress_sequence counter
couch2pg_progress_sequence{cht_instance="192-168-68-26.local-ip.medicmobile.org",db="medic",job="db_targets",target="db1"} 221
# HELP couch2pg_up 1 if couch2pg is running and has updated in the last minute, 0 if not
# TYPE couch2pg_up gauge
couch2pg_up{cht_instance="192-168-68-26.local-ip.medicmobile.org",db="medic",job="db_targets",target="db1"} 1
# HELP dbt_execution_time dbt run last execution time (ms)
# TYPE dbt_execution_time gauge
dbt_execution_time{job="db_targets",table_name="contact",target="db1"} 0.09859967231750488
dbt_execution_time{job="db_targets",table_name="contact_type",target="db1"} 0.07459473609924316
dbt_execution_time{job="db_targets",table_name="data_record",target="db1"} 0.10951995849609375
dbt_execution_time{job="db_targets",table_name="dbt_results",target="db1"} 0.23592472076416016
dbt_execution_time{job="db_targets",table_name="document_metadata",target="db1"} 0.2093191146850586
dbt_execution_time{job="db_targets",table_name="patient",target="db1"} 0.0865786075592041
dbt_execution_time{job="db_targets",table_name="person",target="db1"} 0.10456228256225586
dbt_execution_time{job="db_targets",table_name="place",target="db1"} 0.16501379013061523
dbt_execution_time{job="db_targets",table_name="user",target="db1"} 0.0803828239440918
# HELP dbt_latency difference between last timestamp in dbt models and current time (seconds)
# TYPE dbt_latency gauge
dbt_latency{job="db_targets",target="db1"} 1180.199879
# HELP scrape_duration_seconds How long it took to scrape the target in seconds
# TYPE scrape_duration_seconds gauge
scrape_duration_seconds{job="db_targets",target="db1"} 0.000738378
# HELP up 1 if the target is reachable, or 0 if the scrape failed
# TYPE up gauge
up{job="db_targets",target="db1"} 1
```

that all said, my couch2pg backlog panel in Grafana shows no data:

![image](https://github.com/user-attachments/assets/f7a60157-9e28-4ed3-8458-7360e5a8f57c)

Reply:
@mrjones-plip I suspect the first error you got is because your `POSTGRES_TABLE` env variable was set to medic but that was updated to `couchdb` in this [commit](https://github.com/medic/cht-sync/commit/af115dd74887876d97584381466250d9f1cb2201). I have managed to replicate the No data issue but I don't know why since the SQL query works as expected. I have booked a 30 minute session in your calendar to go through this together and hopefully get this over the line.